fvm/environment/event_emitter.go

```diff
@@ -82,15 +80,14 @@ func NewEventEmitter(
 		txIndex:                       txIndex,
 		payer:                         payer,
 		serviceEventCollectionEnabled: serviceEventCollectionEnabled,
-		eventCollectionByteSizeLimit:  eventCollectionByteSizeLimit,
 	}
 
 	emitter.Reset()
 	return emitter
 }
 
 func (emitter *eventEmitter) Reset() {
-	emitter.eventCollection = NewEventCollection()
+	emitter.eventCollection = NewEventCollection(emitter.meter)
```

Comment:
This change seems wrong.  In particular, this no longer reset the totalByteSize / eventCounter to zero.  This (potentially) prevent the environment from being reusable since emitting an event after Reset will automatically fail.

Reply:
Good catch - to make sure that EventEmitter still uses the Meter instance from Context, a `Reset()` interface seems to be necessary in `Meter`, in which we may need to distinguish which part of metrics to reset, so that when we do reset in event_emitter, other metrics for storage\computation etc. can remain the same. I'd rather not creating new Meter instance here, so in case MeterParameter changes from Context can also be reflected automatically. WDYT

Reply:
it's kind of weird to have a Reset in meter that reset some, but not all, limits.  We should either reset consistently, or just keep the event limits as a separate thing.

Reply:
given #3104 is also on our list, maybe having a separate meter for event emitter is viable solution for now, which can make things easier when we get to splitting existing meter into sub-parts afterwards. @janezpodhostnik

Reply:
o, you know what, i totally missed the enabledLimits check in environment/meter.go .   I think what you're doing is actually ok, as long as we're happy with not checking the emitted event size when limits are disabled.  I don't have any objections to your changes, but we should double check with @janezpodhostnik cuz i don't really know why we have this limit in the first place =P

Reply:
Yeah since we disable the metering after first failure, for fee deduction, this should be fine. We don't reset the meter on other stuff that we reset either.

The limit is here because emitting to many events causes problems with getting the block event list through the API. In the past blocks with large event lists were unfetchable. Also I think because of storage.